src/audio/copier/copier.c

```diff
@@ -1003,9 +1003,14 @@ static int copier_params(struct comp_dev *dev, struct sof_ipc_stream_params *par
 	if (cd->endpoint_num) {
 		for (i = 0; i < cd->endpoint_num; i++) {
 			update_internal_comp(dev, cd->endpoint[i]);
-			ret = cd->endpoint[i]->drv->ops.params(cd->endpoint[i], params);
-			if (ret < 0)
-				break;
+			if (!cd->endpoint[i]->drv->ops.params) {
+				comp_err(dev, "NULL .params for driver %p",
+					 cd->endpoint[i]->drv);
```

Comment:
no need to break and return an error here?

Reply:
@ranj063 I think the NULL dereference is elsewhere. I've just got another exception locally, it's again in `copier_params()` but I think it's the `.drv` pointer above that is invalid. This time it was equal to 0x60d39, any ideas?

Reply:
@lyakh array handling/pointer arithmetic could be a problem here with cd->endpoint[] ?

Reply:
> @lyakh array handling/pointer arithmetic could be a problem here with cd->endpoint[] ?

it *should* be correct - the array should have `cd->endpoint_num` elements. I'm thinking about some race between copying and freeing. Will investigate today